src/main/java/com/google/api/codegen/discovery/DefaultString.java

```diff
@@ -96,7 +75,7 @@ public static String getNonTrivialPlaceholder(String pattern, String format) {
     return "";
   }
 
-  private static final String WILDCARD_PATTERN = "[^/]*";
+  private static final Set<String> WILDCARD_PATTERNS = ImmutableSet.of("[^/]+", ".+", ".*");
```

Comment:
I had excluded the latter because it doesn't appear in any discovery docs anymore, but just in case it makes a comeback I've put it in.